WebAPI/Controllers/OrganizationsController.cs

```diff
@@ -0,0 +1,139 @@
+using EntityModel;
+using Microsoft.AspNetCore.Mvc;
+using Shared;
+using Shared.ApiInterface;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using WebAPI.Models;
+
+namespace WebAPI.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class OrganizationsController : ControllerBase
+    {
+        protected readonly IApiInterface api;
+
+        public OrganizationsController(EfInterface api) => this.api = api ?? throw new ArgumentNullException(nameof(api));
+
+        // GET: api/organizations?name={}&lat={}&lon={}&services={}&distance={}
+        [HttpGet]
+        public async Task<ActionResult<List<OrganizationDTO>>> Get(string name = "", string lat = "", string lon = "", string services = "", double maxDistance = 25)
+        {
+            var organizations = await api.GetVerifiedOrganizations();
+
+
+            if (organizations != null)
+            {
+                var orgsDTO = new List<OrganizationDTO>();
+
+                if (!String.IsNullOrEmpty(name))
+                {
+                    organizations = organizations.Where(o => o.Name == name).ToList();
+                }
+                if (!(String.IsNullOrEmpty(lat) && String.IsNullOrEmpty(lon)))
+                {
+                    Coordinates searchCoordinates = new Coordinates(Convert.ToDouble(lat), Convert.ToDouble(lon));
+
+                    List<Organization> tempList = new List<Organization>();
+
+                    foreach (Organization org in organizations)
+                    {
+                        Coordinates organizationCoordinates = new Coordinates(Convert.ToDouble(org.Latitude), Convert.ToDouble(org.Longitude));
+                        var distanceTo = searchCoordinates.DistanceTo(organizationCoordinates, UnitOfLength.Miles);
+                        if (distanceTo < maxDistance) { tempList.Add(org); }
+                    }
+                    organizations = tempList;
+                }
+                if (!String.IsNullOrEmpty(services))
+                {
+                    List<Organization> tempList = new List<Organization>();
+                    foreach (Organization org in organizations)
+                    {
+                        var ServicesProvided = await api.GetServices(org.Id);
+                        var ServicesName = ServicesProvided.Select(s => s.Name).ToList();
+                        var retServices = ServicesName.Intersect(services.Split(",")).Any();
+
+                        if (retServices)
+                        {
+                            tempList.Add(org);
+                        }
+                    }
+                    organizations = tempList;
+                }
+
+                foreach (var org in organizations)
+                {
+                    var ServicesProvided = await api.GetServices(org.Id);
```

Comment:
It looks like you grab the services in the foeach loop above, so you could store those, that way you don't have to fetch them from the DB again here